lib/smart_proxy_container_gateway/container_gateway_main.rb

```diff
@@ -4,11 +4,13 @@
 require 'sequel'
 module Proxy
   module ContainerGateway
+    extend Proxy::DHCP::DependencyInjection
     extend ::Proxy::Util
     extend ::Proxy::Log
 
+    inject_attr :database, :database
```

Comment:
This provides a property `database` on an instance, but I see everything is static. It would be better to refactor the class so it can be instantiated and inserted via dependency injection.

Another note: `database` is an instance of `Proxy::ContainerGateway::Database`, so you may need to use `database.connection` (and expose that) or change the dependency injection to provide a `Sequel::Database` instance.